src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+const listItem = document.querySelectorAll('li');
+
+function sortList(list) {
+  const arrayList = Array.from(list);
+
+  const sortedList = arrayList.sort((a, b) => {
+    const salaryA = Number(a.dataset.salary.replace(/[^0-9.-]+/g, ''));
+    const salaryB = Number(b.dataset.salary.replace(/[^0-9.-]+/g, ''));
+
+    return salaryB - salaryA;
+  });
+
+  const parents = document.querySelector('ul');
```

Comment:
Consider checking if the `parents` variable is not null before proceeding. This will prevent potential runtime errors if the `ul` element is not found.